pkg/k8s/service.go

```diff
@@ -444,9 +444,22 @@ func NewService(ips []net.IP, externalIPs, loadBalancerIPs, loadBalancerSourceRa
 		loadBalancerSourceCIDRs[cidr.String()] = cidr
 	}
 
+	// If EnableNodePort is not true we do not want to process
+	// events which only differ in external or load balancer IPs.
+	// By omitting these IPs in the returned Service object, they
+	// are no longer considered in equality checks and thus save
+	// CPU cycles processing events Cilium will not act upon.
 	if option.Config.EnableNodePort {
```

Comment:
Have you considered completely removing the guard?

Reply:
yes @aanm opt'd keep to to avoid excess CPU processing. 

lets chat about this here. I'm not sure if any actual measuring was done to prove this necessary. 

To me, I don't like that we have a data model with fields that change based on configuration. I'd much rather see a consistent data model,  and the watcher code "filter" out events based on config, not alter a data model to perform that filtering.

Reply:
SGTM.